### PR TITLE
Bump Dendrite revisions to remove last reference to port-based API

### PR DESCRIPTION
**Pull request by bnaecker**

### CODE REVIEW

**Comment by bnaecker** [comment]
This is a WIP, the last in a chain of PRs integrating support for a new transceiver-controller protocol. See https://github.com/oxidecomputer/transceiver-control/pull/71 for details on the protocol updates and motivation. The series is:

1. https://github.com/oxidecomputer/transceiver-control/pull/71
2. https://github.com/oxidecomputer/hubris/pull/1239
3. https://github.com/oxidecomputer/dendrite/pull/192
4. This PR

This is mostly up now as a TODO and to run CI. I've passed the tests locally, and will revise the actual commits once the Dendrite PR merges to `main`.

File: tools/scrimlet/softnpu-init.sh [hunk 1]
```diff
@@ -13,27 +13,29 @@ echo "Using $GATEWAY_IP as gateway ip"
 GATEWAY_MAC=${GATEWAY_MAC:=$(arp "$GATEWAY_IP" | awk -F ' ' '{print $4}')}
 echo "Using $GATEWAY_MAC as gateway mac"
 
-# Sidecar Interface facing "sled"
-./out/softnpu/swadm -h "[fd00:1122:3344:101::2]" port create 1:0 100G RS
+SWADM="./out/softnpu/swadm -h [fd00:1122:3344:101::2]"
```
Comment (rcgoodfellow) [inline tools/scrimlet/softnpu-init.sh:16]: Just a quick note on this. Based on @Nieuwejaar's comment [here](https://github.com/oxidecomputer/omicron/pull/2645#discussion_r1157531870), we're moving away from using an ad-hoc download of `swadm` and using the `swadm` located in the `oxz_switch` zone via `zlogin`. This is implemented in both #2645 and #2694 so depending on how this PR gets ordered in with those, we'll want to be careful on rebase/merge to preserve the `zlogin` based scheme.

**Comment by bnaecker** [comment]
Gentle ping on this @internet-diglett!

File: tools/scrimlet/softnpu-init.sh (hunk 1 above)
Comment (bnaecker) [reply tools/scrimlet/softnpu-init.sh]: Rebased on changes that include #2694, thanks!

**Comment by bnaecker** [comment]
@internet-diglett This has been updated and reduced to just moving the softnpu init script over to `swadm link`. I'd love a review when you get a moment, so I can close out https://github.com/oxidecomputer/dendrite/pull/194

File: tools/scrimlet/softnpu-init.sh [hunk 2]
```diff
@@ -37,7 +36,7 @@ z_swadm arp add "$GATEWAY_IP" "$GATEWAY_MAC"
 # Configure route to upstream gateway
 z_swadm route add 0.0.0.0/0 qsfp0/0 "$GATEWAY_IP"
 
-z_swadm port list
+z_swadm link ls
```
Comment (internet-diglett) [inline tools/scrimlet/softnpu-init.sh:39]: Not a big problem, but just curious why we are using `ls` as a subcommand for `link`, but the other commands are still using `list`?

**Review by internet-diglett — COMMENTED** [review]
Do we need to update the dependencies in `package-manifest.toml` for these commands to work? (the commands that have been updated are executed via the switch zone, so the switch zone would need a newer version of swadm, right?)

File: tools/scrimlet/softnpu-init.sh (hunk 2 above)
Comment (internet-diglett) [reply tools/scrimlet/softnpu-init.sh:39]: Ah, nevermind, I just launched the zone to test and found:

```
root@oxz_switch:~# /opt/oxide/dendrite/bin/swadm link
swadm-link 0.1.0
Manage Ethernet links.

Links are always referenced to the switch port that contains them. This is normally done by using a "link path", which
is structured like `switch_port/link_id`, e.g., `rear0/0`.

USAGE:
    swadm link <SUBCOMMAND>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

SUBCOMMANDS:
    create      Create a new link on a switch port
    delete      Delete a link on a switch port [aliases: del]
    get         Get a single link
    get-prop    Get a property of a link
    help        Prints this message or the help of the given subcommand(s)
    list        List all links on a switch port
    list-all    List all links on all switch ports [aliases: ls] <-- tada! 🎉 
    set-prop    Set a property of a link
```

**Review by internet-diglett — APPROVED** [review]
🚢

File: tools/scrimlet/softnpu-init.sh (hunk 2 above)
Comment (bnaecker) [reply tools/scrimlet/softnpu-init.sh:39]: Just an alias yeah.

**Pull request merged**